### ISSUE TITLE
Broken division operation on Duration objects

### ISSUE THREAD

**Issue by Sayanc93** [rails/rails#29952]
Division operation on Duration objects is broken in 5.2.0.alpha

```ruby
4.minutes / 2.minutes
=> 0 minutes

4.minutes / 1.minute
=> 0 minutes
```

While, `(4.minutes / 2.minute).value` does yield `4` but returning `0 minutes` is incorrect representation.
Also, notice that

```ruby
time = 4.minutes / 1.minute
=> 4 minutes

time / 2.minutes
=> 0.minutes

time.value
=> 4

2.minutes.value
=> 120
```

Chaining of operations on Duration objects fails due to inconsistencies in value attribute.

The values should be consistent with the protocol set [here](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/duration.rb#L143-L169) to allow operation chaining on Duration objects i.e set values in seconds.

**Comment by fphilipe** [comment]
I've experienced this behavior as well. It used to be that the time units would cancel out on division. Just checked and this changed starting in version 5.1.0.

The following used to be the case before 5.1.0:

```
irb(main):001:0> require 'active_support/all'
=> true
irb(main):002:0> 1.hour / 1.minute
=> 60
```

Since 5.1.0 this is the observed behavior:

```
irb(main):001:0> require 'active_support/all'
=> true
irb(main):002:0> 1.hour / 1.minute
=> 0 hours
irb(main):003:0> (1.hour / 1.minute).to_i
=> 60
```

**Comment by pixeltrix** [comment]
Fixed by #29971
